### PR TITLE
Feat - improve

### PR DESCRIPTION
**Pull request by biswanath-tewari**
fixes:-
1. blurred loading image
2. filter by a category
Feature:-
1. added basic animations

### CODE REVIEW

**Review by Prabhav-Sharma — APPROVED** [review]

File: src/hooks/useScroll.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { useInView } from 'react-intersection-observer'
+import { useAnimation } from 'framer-motion'
+
+const useScroll = () => {
+  const controls = useAnimation() // animation
+  const [element, view] = useInView({ threshold: 0.3 })
+
+  // if element is in view, we are starting the show animation
```
Comment (prathmeshjagtap) [inline src/hooks/useScroll.js:8]: remove commented code

**Review by prathmeshjagtap — APPROVED** [review]
Everything else is good

File: src/hooks/useScroll.js (hunk 1 above)
Comment (biswanath-tewari) [reply src/hooks/useScroll.js:8]: that is a comment describing what the code is doing.

File: src/utils/animate.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    y: 300,
+  },
+
+  show: {
+    opacity: 1,
+    y: 0,
+    transition: {
+      duration: 1,
+      when: 'beforeChildren',
+      staggerChildren: 0.25,
```
Comment (shivanigangadharan) [inline src/utils/animate.js:13]: I like this transition effect used here.

**Review by shivanigangadharan — APPROVED** [review]
Code looks good, implemented nice UI features. Can merge.

**Pull request merged**